tuf/tuf.go

```diff
@@ -173,6 +173,132 @@ func (tr *Repo) RemoveBaseKeys(role string, keyIDs ...string) error {
 	return nil
 }
 
+// GetBaseRole gets a base role from this repo's metadata
+func (tr *Repo) GetBaseRole(name string) (data.BaseRole, error) {
+	if !data.IsBaseRole(name) {
+		return data.BaseRole{}, fmt.Errorf("Invalid base role name specified: %s", name)
+	}
+	roleData, ok := tr.Root.Signed.Roles[name]
+	if !ok {
+		return data.BaseRole{}, fmt.Errorf("TUF data for %s does not exist in root metadata", name)
```

Comment:
Non-blocking - I'm not sure if this is the right error, but would `ErrNotLoaded` be useful for this?  We seem to currently be using it for returning an error if the actual `SignedX` stuff isn't loaded, but the error message says "role".